cluster-autoscaler/cloudprovider/aws/aws_manager_test.go

```diff
@@ -30,6 +33,32 @@ import (
 	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
 )
 
+// TestGetRegion ensures correct source supplies AWS Region.
+func TestGetRegion(t *testing.T) {
+	key := "AWS_REGION"
+	originalRegion, originalPresent := os.LookupEnv(key)
+	defer func(region string, present bool) {
+		os.Unsetenv(key)
+		if present {
+			os.Setenv(key, region)
+		}
+	}(originalRegion, originalPresent)
+	// Ensure environment variable retains precedence.
+	expected1 := "the-shire-1"
+	os.Setenv(key, expected1)
+	assert.Equal(t, expected1, getRegion())
+	// Ensure without environment variable, EC2 Metadata used... and it merely
+	// chops the last character off the Availability Zone.
+	expected2 := "mordor-2"
```

Comment:
The AWS SDK’s formula for retrieving the region is to actually to get the Availability Zone and remove its last character:

https://github.com/aws/aws-sdk-go/blob/master/aws/ec2metadata/api.go#L115

Since this test fixture mimics that HTTP call to get the Availability Zone, an extra character has to be present.